pkg/resolve/remote.go

```diff
@@ -32,74 +31,161 @@ func alreadySeen[T NamedItem](items []T, item T) bool {
 //
 // The precedence logic for Pipeline is first from PipelineRun annotations and
 // then from Tekton directory.
-func getRemotes(ctx context.Context, rt *matcher.RemoteTasks, types TektonTypes) (TektonTypes, error) {
-	remoteType := &TektonTypes{}
+func resolveRemoteResources(ctx context.Context, rt *matcher.RemoteTasks, types TektonTypes, ropt *Opts) ([]*tektonv1.PipelineRun, error) {
```

Comment:
This function is too big can we split into smaller function specific to task and pipeline